cmake/luajit.cmake

```diff
@@ -69,14 +69,15 @@ endif()
 # Enable LuaJIT ASan support. The internal LuaJIT memory allocator
 # is not instrumented yet unfortunately, so to find any memory
 # faults it's worth building LuaJIT with system provided memory
-# allocator (i.e. enable LUAJIT_USE_SYSMALLOC option). However,
-# Tarantool doesn't finalize Lua universe the right way (see the
-# comments near <tarantool_lua_free>), so running Tarantool
-# testing routine with LUAJIT_USE_SYSMALLOC enabled generates
-# false-positive LSan leaks. Return back here to enable
-# LUAJIT_USE_SYSMALLOC, when the issue below is resolved.
-# https://github.com/tarantool/tarantool/issues/3071
+# allocator (i.e. enable LUAJIT_USE_SYSMALLOC option). Also we need
+# to set LUAJIT_ENABLE_GC64 because for non-GC64 builds we can't
+# use sysmalloc since the bitness of the allocator and pointers should
+# be the same.
 if(ENABLE_ASAN)
+    set(LUAJIT_ENABLE_GC64 ON CACHE BOOL
```

Comment:
I suppose we shouldn't silently enable GC64 when the corresponding flag isn't set.
I think we should enable `LUAJIT_USE_SYSMALLOC` only for GC64 build and raise a _warning_ otherwise.

Reply:
Okay, `LUAJIT_USE_SYSMALLOC` is set only when `GC64=ON`.